src/webpage-digester.ts

```diff
@@ -1,16 +1,27 @@
 import {IndexFilter} from "./filters/index-filter";
 import {Filter} from "./filters/filter";
 import {WebPage} from "./utils/webpage";
+import {Term} from "./utils/term";
+import {Occurrence} from "./utils/occurrence";
 
 
 export class WebPageDigester {
     private searchTerms : Set<string>;
+    private termToIDMap : Map<string, string>;
     private mainFilter : new (searchTerms? : Set<string>) =>  IndexFilter;
     private mainFilterInstance : IndexFilter;
     private preFilterInstance : Filter;
 
-    constructor(searchTerms : Array<string>) {
-        this.searchTerms = new Set(searchTerms);
+    constructor(searchTerms : Array<Term>) {
+        this.searchTerms = new Set();
+        this.termToIDMap = new Map();
```

Comment:
You're storing the search terms twice. Please get rid of the `Set` and only use the `Map` as we only want to keep the terms once in memory. We only need the `Set` for initializing `Filter`s which shouldn't happen so often. So creating a `Set` from the `Map` for initializing `Filters` shouldn't be a big performance hit at all.